WordPress/Docs/WP/GetMetaFunctionSingleParameterStandard.xml

```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0"?>
+<documentation xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:noNamespaceSchemaLocation="https://phpcsstandards.github.io/PHPCSDevTools/phpcsdocs.xsd"
+    title="Get Meta Function Single Parameter"
+    >
+    <standard>
+    <![CDATA[
+    When calling the `get_(comment|post|site|term|user)_meta()`, `get_metadata()`,
+    `get_metadata_default()`, and `get_metadata_raw()` functions, if the `$[meta_]key` parameter is
+    passed, the `$single` parameter should be passed as well to make it explicit what is the
+    expected return type.
+    ]]>
```

Comment:
Good point! I agree that it is worth including a note about this. That being said, unless I'm missing something, I don't think `false` is a valid value for a single meta value. WP converts `false` to an empty string when adding a new meta to the database.

> An empty string if a valid but non-existing comment ID is passed.

Besides that, the documentation for the get meta functions is incorrect as far as I can check. They may return an empty string if a valid but non-existing ID is passed only when `$single` is `true`. If `$single` is `false`, they may return an empty array. I can open Trac ticket to update the documentation.

https://github.com/WordPress/wordpress-develop/blob/4f56c3d18a7236048ed40bfa8c82fb8c88f9b361/src/wp-includes/meta.php#L673-L674

https://github.com/WordPress/wordpress-develop/blob/4f56c3d18a7236048ed40bfa8c82fb8c88f9b361/src/wp-includes/meta.php#L688-L692

Considering the above, I believe we could add the following paragraph:

"Note that for a valid but non-existing ID these functions may return an empty string if `$single` is `true` or an empty array if `$single` is `false`. This can be confusing as these might be valid values for a meta value."

What do you think?

Reply:
I agree that opening a Trac ticket to fix the return value documentation might be a good idea.

Then again, this could probably be fixed via the catch-all [WP docs Trac ticket for WP 6.7](https://core.trac.wordpress.org/ticket/61608).

Probably simplest to just create a patch, link it to that ticket and then leave a comment on the Trac ticket so people get notified to take a look.

Reply:
> Good point! I agree that it is worth including a note about this. That being said, unless I'm missing something, I don't think `false` is a valid value for a single meta value. WP converts `false` to an empty string when adding a new meta to the database.

To be honest, that sounds like a bug in WP (or in one of the filters hooked into something called within the `add/update_metadata` functions, as the documentation says that the type of `$value` is `mixed` and doesn't document any restrictions on `mixed` (other than non-scalars being serializable). If the type of the `$value` gets changed in some way, that will lead to lots of WTF situations, so this is problematic.

Would be worth looking into, but not something we need to address in this PR (outside scope).

If recently, it should be fixed as a bug fix.
If this type conversion has been happening for quite a while, fixing it would be a BC-break, so can't be done. In that case, the documentation for `add/update/get_metadata` (and all the variants) should be fixed to document the type casts/juggling being done.

Also would be good to add (probably a sh*tload of) test to document the behaviour and prevent it getting worse in the future ;-)


> Considering the above, I believe we could add the following paragraph:
> 
> "Note that for a valid but non-existing ID these functions may return an empty string if `$single` is `true` or an empty array if `$single` is `false`. This can be confusing as these might be valid values for a meta value."
> 
> What do you think?

Sounds like a good addition.

Reply:
> I agree that opening a Trac ticket to fix the return value documentation might be a good idea.
> Then again, this could probably be fixed via the catch-all [WP docs Trac ticket for WP 6.7](https://core.trac.wordpress.org/ticket/61608).
> Probably simplest to just create a patch, link it to that ticket and then leave a comment on the Trac ticket so people get notified to take a look.

Here is the comment on the ticket that you mentioned with a link to the PR: https://core.trac.wordpress.org/ticket/61608#comment:9

Reply:
> If recently, it should be fixed as a bug fix.
> If this type conversion has been happening for quite a while, fixing it would be a BC-break, so can't be done. In that case, the documentation for add/update/get_metadata (and all the variants) should be fixed to document the type casts/juggling being done.

I just created another ticket for this other issue: https://core.trac.wordpress.org/ticket/61950

Reply:
> Here is the comment on the ticket that you mentioned with a link to the PR: https://core.trac.wordpress.org/ticket/61608#comment:9

That is the automated PR link comment. People don't get notified about those. You need to also leave an actual comment on the Trac ticket to say that a new PR has been added to the ticket (and get the notifications send out).

Reply:
Thanks, @jrfnl. I was not aware of that. I assumed that an actual comment was not necessary as in that same ticket, other contributors left just the automated PR link comment, and that was enough for some WP core developer to review the PR. I will leave a comment tomorrow.

Reply:
Thanks! :raised_hands: